apps/src/templates/instructions/TopInstructionsCSF.jsx

```diff
@@ -193,12 +194,21 @@ var TopInstructions = React.createClass({
   adjustMaxNeededHeight() {
     const minHeight = this.getMinHeight();
 
-    const instructionsContent = this.refs.instructions;
-    const maxNeededHeight = $(ReactDOM.findDOMNode(instructionsContent)).outerHeight(true) +
-      RESIZER_HEIGHT;
+    if (this.props.hasContainedLevels) {
+      const maxNeededContainedLevelHeight =
+        $(ReactDOM.findDOMNode(this)).find('#containedLevelContainer').outerHeight(true) +
+        RESIZER_HEIGHT;
+
+      this.props.setInstructionsMaxHeightNeeded(Math.max(minHeight, maxNeededContainedLevelHeight));
+      return maxNeededContainedLevelHeight;
+    } else {
+      const instructionsContent = this.refs.instructions;
+      const maxNeededHeight = $(ReactDOM.findDOMNode(instructionsContent)).outerHeight(true) +
+        RESIZER_HEIGHT;
 
-    this.props.setInstructionsMaxHeightNeeded(Math.max(minHeight, maxNeededHeight));
-    return maxNeededHeight;
+      this.props.setInstructionsMaxHeightNeeded(Math.max(minHeight, maxNeededHeight));
+      return maxNeededHeight;
+    }
```

Comment:
Not yet - I'm trying to make consistent changes across both because of the way that TopInstructions is sort of both CSF and CSP, but I need to generalize the gamelab part before I can test it in the other mode.